luigi/contrib/ftp.py

```diff
@@ -127,10 +127,14 @@ def _sftp_exists(self, path, mtime):
         return exists
 
     def _ftp_exists(self, path, mtime):
+        path_parts = path.split('/')
+        path = '/'.join(path_parts[:-1])
+        fn = path_parts[-1]
```

Comment:
Does this assignment make sense?? You seem to override it right afterwards in the loop no?